hs_core/serialization.py

```diff
@@ -718,32 +716,19 @@ def write_metadata_to_resource(self, resource):
                           'phone': c.phone, 'homepage': c.homepage,
                           'researcherID': c.researcherID,
                           'researchGateID': c.researchGateID}
-                try:
-                    resource.metadata.create_element('contributor', **kwargs)
-                except IntegrityError:
-                    pass
+                resource.metadata.create_element('contributor', **kwargs)
             else:
                 msg = "Contributor with type {0} are not supported"
                 msg = msg.format(c.__class__.__name__)
                 raise TypeError(msg)
         if self.abstract:
-            try:
-                resource.metadata.create_element('description', abstract=self.abstract)
-            except IntegrityError:
-                resource.metadata.update_element('description', resource.metadata.description.id,
-                                                 abstract=self.abstract)
+            resource.metadata.create_element('description', abstract=self.abstract)
```

Comment:
@pkdash When the resource is created, there is no description (i.e. abstract), this only gets created by write_metadata_to_resource().  I believe the former code was being overly cautious to avoid errors, but I do not believe this error (the description already existing) could conceivably happen given that the same body of code controls the creation of the resource.